WordPress/Classes/Services/PostService.m

```diff
@@ -259,11 +259,17 @@ - (void)uploadPost:(AbstractPost *)post
             Post *postInContext = (Post *)[self.managedObjectContext existingObjectWithID:postObjectID error:nil];
             if (postInContext) {
                 postInContext.remoteStatus = AbstractPostRemoteStatusFailed;
-                // If the post was not created on the server yet we convert the post to a local draft post with the current date.
-                if (!postInContext.hasRemote) {
+                // If the post was not created on the server yet we convert the post to a local
+                // draft post with the current date. However, if it's PUBLISHED or DRAFT, we will
+                // not reset the status so that when retrying the upload, the post will have the
+                // status that the user originally intended.
+                //
+                // When we support auto-uploading of all statuses, we will fully remove this block.
+                if (!postInContext.hasRemote && ![postInContext.status isEqualToString:PostStatusPublish]) {
                     postInContext.status = PostStatusDraft;
```

Comment:
Apologize if this is a dumb question, but I'm confused how this line (setting `postInContext.status` to `draft`  if a upload failure occured, is supposed to work with the new filtering on `getPostsToRetry`. 

If the only statuses a failed post can have is either:
a) `publish`, because we explicitly check for it and don't touch it
b) `draft`, because we blindly overwrite it in every other case

...then isn't the filtering on `getPostsToRetry()` kinda pointless?

Reply:
Great observation! And it's not a dumb question. You're absolutely right. As it is right now, it does look pointless. 

In the next following PRs, we will be adding safety features like: 

- Do not auto-upload if the failure was more than 48 hours ago. Reference: https://github.com/wordpress-mobile/WordPress-Android/issues/9555#issuecomment-510844042 
- Do not auto-upload if the user has not confirmed intent to publish https://github.com/wordpress-mobile/WordPress-iOS/pull/12252#issuecomment-517480673. 
- Possibly remote auto-save instead if we cannot auto-upload.

These are currently proposals. We'd love to hear what you think about those. 

At a minimum, `FailedPostsFetcher` will look like this in the future:

```swift
postService.getFailedPosts { posts in
    let postsToRetry = posts.filter({ post -> Bool in
        guard let status = post.status else {
	    return false
        }
    
        if failureHappenedMoreThan48HoursAgo() {
    	    return false
        }
    
        if userDidNotConfirmIntent() {
    	    return false
        }
    
        // This will no longer be needed if we support all statuses
        // return allowedStatuses.contains(status) && !post.hasRemote()

        return true
    })
    
    result(postsToRetry)
}
```

Reply:
alrighty, with more context that makes much more sense — i was being perhaps overly cautious due to the recent problems ;) thanks for the explanation!